File: examples/nodejs/01_basic_kv.js
const Redis = require("ioredis");

// Connect to Zedis
const redis = new Redis({
  host: "localhost",
  port: 6379,
});

async function main() {
  console.log("--- 01 Basic Key-Value Operations ---");

  // SET
  console.log("Setting key 'app:status' to 'running'...");
  await redis.set("app:status", "running");

  // GET
  const status = await redis.get("app:status");
  console.log(`Got value: ${status}`);

  // EXPIRY
  console.log("Setting key 'session:temp' with 10s expiry...");
  await redis.set("session:temp", "xyz", "EX", 10);
  const ttl = await redis.ttl("session:temp");
  console.log(`TTL: ${ttl}`);

  // PING
  const pong = await redis.ping();
  console.log(`PING response: ${pong}`);

  redis.disconnect();
}

main();

File: examples/python/01_basic_kv.py
import redis
import time

# Connect to Zedis
r = redis.Redis(host='localhost', port=6379, decode_responses=True)

print("--- 01 Basic Key-Value Operations ---")

# SET
print("Setting key 'user:1:name' to 'Alice'...")
r.set('user:1:name', 'Alice')

# GET
name = r.get('user:1:name')
print(f"Got value: {name}")

# EXPIRY
print("Setting key 'temp:otp' with 5s expiry...")
r.setex('temp:otp', 5, '123456')
ttl = r.ttl('temp:otp')
print(f"TTL: {ttl}")

# INCR
r.set('counter', 0)
new_val = r.incr('counter')
print(f"Counter incremented to: {new_val}")
